src/cim/components/ClusterDeployment/ClusterDeploymentDetails.tsx

```diff
@@ -0,0 +1,242 @@
+        const kubeconfig = kubeconfigSecret.data?.kubeconfig;
+
+        if (!kubeconfig) throw new Error('Kubeconfig is empty.');
+
+        const blob = new Blob([atob(kubeconfig)], { type: 'text/plain;charset=utf-8' });
+        saveAs(blob, 'kubeconfig.json');
+      } catch (e) {
+        console.error('Failed to fetch kubeconfig secret.', e);
+      }
+    }
+  };
+
+  const handleFetchEvents: EventListFetchProps['onFetchEvents'] = async (
+    props,
+    onSuccess,
+    onError,
+  ) => {
+    try {
+      const eventsURL = agentClusterInstall.status?.debugInfo?.eventsURL;
+      if (!eventsURL) throw 'Events URL is not available.';
+
+      const res = await fetch(eventsURL);
+      const rawData: Record<string, string>[] = await res.json();
+      const data = formatEventsData(rawData);
+
+      onSuccess(data);
+    } catch (e) {
+      onError('Failed to fetch cluster events.');
+    }
+  };
+
+  const cluster = getAICluster({ clusterDeployment, agentClusterInstall, agents });
+  const [clusterStatus, clusterStatusInfo] = getClusterStatus(agentClusterInstall);
+  return (
+    <Stack hasGutter>
+      {shouldShowClusterInstallationProgress(agentClusterInstall) && (
+        <StackItem>
+          <Card id="cluster-installation-progress-card" isExpanded={progressCardExpanded}>
+            <CardHeader
+              onExpand={() => setProgressCardExpanded(!progressCardExpanded)}
+              toggleButtonProps={{
+                id: 'progress-card-toggle-button',
+                'aria-label': 'Cluster installation process',
+                'aria-labelledby': 'titleId progress-card-toggle-button',
+                'aria-expanded': progressCardExpanded,
+              }}
+            >
+              <CardTitle id="titleId">Cluster installation process</CardTitle>
+            </CardHeader>
+            <CardExpandableContent>
+              <CardBody>
+                <Stack hasGutter>
+                  <StackItem>
+                    <ClusterDeploymentProgress
+                      clusterDeployment={clusterDeployment}
+                      agentClusterInstall={agentClusterInstall}
+                      agents={agents}
+                      onFetchEvents={handleFetchEvents}
+                    />
+                  </StackItem>
+                  {shouldShowClusterCredentials(agentClusterInstall) && (
+                    <StackItem>
+                      <ClusterDeploymentCredentials
+                        clusterDeployment={clusterDeployment}
+                        agentClusterInstall={agentClusterInstall}
+                        agents={agents}
+                        fetchSecret={fetchSecret}
+                        consoleUrl={
+                          clusterDeployment.status?.webConsoleURL ||
+                          `https://console-openshift-console.apps.${clusterDeployment.spec?.clusterName}.${clusterDeployment.spec?.baseDomain}`
```

Comment:
nit: What about introducing a utility function for that?

Reply:
done